navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java

```diff
@@ -548,6 +596,8 @@ public OffRoute getOffRouteEngine() {
 
   @Override
   public void onProgressChange(Location location, RouteProgress routeProgress) {
+    sessionState = sessionState.toBuilder().arrivalTimestamp(new Date()).build();
+    NavigationMetricsWrapper.arriveEvent(sessionState, routeProgress, location);
```

Comment:
Do we need a check for arrival for the two lines here?

Reply:
this is currently tied into the 0.99% complete (only time this callback gets invoked). Opened https://github.com/mapbox/mapbox-navigation-android/issues/217 to change the percentage btw.